### ISSUE TITLE
package request: xournal

### ISSUE THREAD

**Issue by Rmano** [termux/x11-packages#18]
**Package description**
`xournal` is probably the best program available for annotating PDF files. I tried zillions of native android apps, but no one has the capabilities of `xournal` (for example the automatic recognition of geometric figures) and such a clean annotation code (the annotations are not in bitmap form but in a svg-like fashion).  

**Link to home page and sources**
1. Home page:  http://xournal.sourceforge.net/
2. Source code: the best version is this one: https://github.com/dmgerman/xournal 

Thanks!

**Comment by Rmano** [comment]
One of the problem of `xournal` is that it is GTK2-only. There is a version (very actively developed lately) which is GTK3, but needs g++ to compile. It's largely equivalent if not better...

Source code ans  home page: https://github.com/xournalpp/xournalpp

**Comment by Rmano** [comment]
I tested it; it works (barring problems with the pen that are more dependent on the VNC viewer and7or Xserver), but it is prone to crash the whole termux session. Is there a guide/hint on how can I try to debug it?

**Comment by geometryof** [comment]
Works for me -- almost.  All icons are missing, and there is an error message on startup:

Failed to load pixbuf file: /data/data/com.termux/files/usr/share/xournal/pixmaps/xournal.png: Couldn’t recognize the image file format for file “/data/data/com.termux/files/usr/share/xournal/pixmaps/xournal.png”

That file and directory are indeed present, and appear to contain the proper PNG files.
I can of course not read /data or /data/data, but everything below  files is rw for user, which ought to be sufficient for xournal.  To the best of my knowledge, my system is up-to-date.  Perhaps a libpng issue?  or gtk??

Thanks for any suggestions!

**Comment by xeffyr** [comment]
@geometryof I can't confirm that. See screenshot from VNC:
![remmina_Быстрое подключение_192 168 1 3:5901_2020816-05226,860691](https://user-images.githubusercontent.com/25881154/90324167-fb62a300-df73-11ea-9e3b-32b00dc514b9.png)

**Comment by geometryof** [comment]
I'm using XSDL.  Screenshot attached.  Same result if I set DISPLAY and start xournal from a Termux window.  (This session has fvwm running in background, but I get the same result without it.)

TD
![xournalerrorC](https://user-images.githubusercontent.com/40740019/90325086-00036900-df2c-11ea-9969-8aa945bbb37b.jpg)

**Comment by geometryof** [comment]
If I start xournal on another machine, the icons show up correctly.  So
it does not appear to be a problem with XSDL.  Is there any chance XSDL
is having trouble accessing the filesystem?  Can you confirm that it's
working for you under XSDL?  Given that your installation is working
correctly with VNC, perhaps it is indeed an update problem -- I did just
update, and now get no errors, but it took a couple of tries.  So I
suppose it's possible something's misconfigured.  But that doesn't yet
seem likely enough to me to be worth reinstalling everything...

Thanks,
	TD

**Comment by geometryof** [comment]
OK; problem solved!  Tracked down the error message and modified the
advice at:
	https://roidelapluie.be/wiki/gdkpixbuf/
I couldn't figure out where the default cache is stored --
using $PREFIX/lib/gdk-pixbuf-2.0/2.10.0 did not work.  So I tried:
	gdk-pixbuf-query-loaders --update-cache
and that did the trick.

I suspect that my long-delayed update indirectly led to this cache not
being properly updated.  A reinstallation of xournal and/or gdk-pixbuf
might well have fixed this problem automatically.  That's what I would
have tried next...

**Comment by xeffyr** [comment]
Yes, pixbuf loaders cache is normally updated during `gdk-pixbuf` installation/update.
